src/kernel/lsm/lsm_kv.rs

```diff
@@ -117,7 +118,7 @@ impl KVStore for LsmStore {
     async fn flush(&self) -> Result<()> {
         let (tx, rx) = oneshot::channel();
 
-        self.compactor_tx.send(CompactTask::Flush(Some(tx)))?;
+        self.compactor_tx.send(CompactTask::Flush(Some(tx))).await?;
```

Comment:
之前没加 await 是不是异步任务直接不等待执行了

Reply:
异步方法必须需要await去执行，这里加await是因为compactor变成有界队列后，方法变成异步了

Reply:
> 之前没加 await 是不是异步任务直接不等待执行了

异步方法都需要await驱动，但是这里加await是因为compactor_tx换成有界队列的sender，方法变成异步的了

Reply:
Attempts to send a message on this UnboundedSender without blocking.
This method is not marked async because sending a message to an unbounded channel never requires any form of waiting. Because of this, the send method can be used in both synchronous and asynchronous code without problems.
If the receive half of the channel is closed, either due to close being called or the UnboundedReceiver having been dropped, this function returns an error. The error includes the value passed to send.
```rust
pub fn send(&self, message: T) -> Result<(), SendError<T>> {
        if !self.inc_num_messages() {
            return Err(SendError(message));
        }

        self.chan.send(message);
        Ok(())
    }
```
我明白了官方解释很明白🥲